File: p/php/hello-discord.php
<?php
require __DIR__ . '/vendor/autoload.php';

use Discord\Discord;

$dotenv = Dotenv\Dotenv::createImmutable(__DIR__);
$dotenv->load();

$GLOBALS['token'] = $_ENV['TOKEN'];
$GLOBALS['prefix'] = $_ENV['PREFIX'];

$discord = new Discord([
    'token' => $token,
]);

$discord->on('ready', function ($discord) {
    echo "Bot is ready!", PHP_EOL;

    // Listen for messages.
    $discord->on('message', function ($message, $discord) {
        $prefix = $GLOBALS['prefix']; // what even is this? it's the meaning of life, the universe, and everything; 42
        $guild = $discord->guilds->get('id', $message->guild_id);

        echo "{$message->author->username}: {$message->content}",PHP_EOL;

        // if message is from a bot odr not in a guild, return
        if ($message->author->bot) {
            return;
        }

        // if message does not start with $prefix, return
        if (substr($message->content, 0, strlen($prefix)) !== $prefix) {
            return;
        }
        
        // get arguments from message
        $args = explode(" ", $message->content);
        $command = substr($args[0], strlen($prefix));
        $args = array_slice($args, strlen($prefix));

        echo $command,PHP_EOL;

        // ping command
        if ($command === 'ping') {
          // reply with Pong! and message arguments
          $message->reply("Pong! ${implode(" ", $args)}");
        }

        // ban command
        if ($command === 'ban') {
          $mentioned = $message->mentions->first();
          $id = $mentioned->id ?: "";
          
          $guild->members->fetch($id)->done(
            function ($member) use ($guild, $message) {
              $member->ban(0, implode(" ", $args))->done(
              function () use ($message, $member) {
                $message->reply("Banned {$member->username}!");
                },
                function ($error) use ($message) {
                  $message->reply("I do not have the required permissions to ban this user!");
                }
              );
            },
            function ($error) use ($message) {
              $message->reply("Please mention a member to ban!");
              return;
            }
          );
        }

        // kick command
        if ($command === 'kick') {
          $mentioned = $message->mentions->first();
          $id = $mentioned->id ?: "";

          // get first mentioned member from collection from message
          $guild->members->fetch($id)->done(
          function ($member) use ($guild, $message) {
              $guild->members->kick($member)->done(
                function () use ($message, $member) {
                  $message->reply("Kicked {$member->username}!");
                },
                function ($error) use ($message) {
                  $message->reply("I do not have the required permissions to kick this user!");
                }
              );
            },
            function ($error) use ($message) {
              $message->reply("Please mention a member to kick!");
              return;
            }
          );
        }

    });
});

$discord->run();

File: p/python/hello-discord.py
# Thanks @Nanofaux!
# (and GitHub Copilot)

import discord
from discord.ext import commands
import os
from dotenv import load_dotenv

load_dotenv()

# to access env vars do it like
env = os.environ.get('key')
bot = commands.Bot(command_prefix=os.environ['PREFIX'])


#kick command with reason argument
@bot.command()
@commands.has_permissions(kick_members=True)
@commands.bot_has_permissions(kick_members=True)
async def kick(ctx, user: discord.Member, *, reason: commands.clean_content = '[No reason given]'): 
    """ Kick a member for being bad """
    author = ctx.author
    guild = ctx.guild
    if user.top_role.position >= ctx.guild.me.top_role.position:
        return await ctx.send("I do not have the required permissions to kick this member.")
    elif user.top_role.position >= author.top_role.position:
        return await ctx.send("I do not have the required permissions to kick this member.")
    try:
        await user.send(f"You were kicked from {guild}, reason: {reason}")
    except discord.Forbidden:
        pass
    await user.kick(reason=f"By {author}, reason: {reason}")


#ban command with reason argument
@bot.command()
@commands.has_permissions(ban_members=True)
@commands.bot_has_permissions(ban_members=True)
async def ban(ctx, user: discord.User, *, reason: commands.clean_content = '[No reason given]'): 
    """ Ban a member for being badd """
    # check if bot has permissions to kick user
    author = ctx.author
    guild = ctx.guild
    if member := guild.get_member(user.id):
        if member.top_role.position >= ctx.guild.me.top_role.position:
            return await ctx.send("I do not have the required permissions to ban this member.")
        elif member.top_role.position >= author.top_role.position:
            return await ctx.send("I do not have the required permissions to ban this member.")
    else:
        # They aren't in the guild obv
        try:
            await guild.fetch_ban(user)
        except discord.NotFound:
            try:
                await user.send(f"You were banned from {guild}, reason: {reason}")
            except discord.Forbidden:
                pass
            await guild.ban(user, reason=f"By {author}, reason: {reason}")
            await ctx.send(f"User {user} has been banned. Reason: {reason}")
        else:
            await ctx.send("That user was already banned from the guild!")


# Ping command
@bot.command()
async def ping(ctx):
    """ Check if bot is online """
    await ctx.send(f"Pong! {round(bot.latency * 1000)}ms")

bot.run(os.environ['TOKEN'])
